war/src/main/js/components/dropdowns/templates.js

```diff
@@ -58,7 +60,7 @@ function menuItem(options) {
           ${label}
                     ${
                       itemOptions.badge != null
-                        ? `<span class="jenkins-dropdown__item__badge" tooltip="${badgeTooltip}">${badgeText}</span>`
+                        ? `<span class="badge alert-${badgeSeverity}" tooltip="${badgeTooltip}">${badgeText}</span>`
```

Comment:
with my proposals this should be
`<span class="jenkins-dropdown__item__badge jenkins-badge alert-${badgeSeverity}" tooltip="${badgeTooltip}">${badgeText}</span>